src/test/java/fr/sncf/osrd/simulation/TrainReactionTest.java

```diff
@@ -0,0 +1,78 @@
+package fr.sncf.osrd.simulation;
+
+import static fr.sncf.osrd.Helpers.*;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import fr.sncf.osrd.infra.InvalidInfraException;
+import fr.sncf.osrd.infra.trackgraph.SwitchPosition;
+import fr.sncf.osrd.infra_state.SwitchState;
+import fr.sncf.osrd.railjson.parser.RailJSONParser;
+import fr.sncf.osrd.railjson.schema.common.ID;
+import fr.sncf.osrd.railjson.schema.infra.railscript.RJSRSExpr;
+import fr.sncf.osrd.railjson.schema.infra.signaling.RJSAspect;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+
+public class TrainReactionTest {
+    @Test
+    public void testWaitingForSwitchChange() throws InvalidInfraException, SimulationError {
+        var infra = getBaseInfra();
+        assert infra != null;
+        infra.switches.iterator().next().positionChangeDelay = 42;
+        var sim = Simulation.createFromInfra(RailJSONParser.parse(infra), 0, null);
+        SwitchState switchState = sim.infraState.getSwitchState(0);
+        switchState.setPosition(sim, SwitchPosition.RIGHT);
+        run(sim);
```

Comment:
Are you sure that the simulation throw an error if the train doesn't wait for the switch?
I think we should add some exception if the switch is moving here:
https://github.com/DGEXSolutions/osrd-core/blob/67bc3ce3ce1b9fb1abadc10071f8592f79703f7e/src/main/java/fr/sncf/osrd/infra_state/SwitchState.java#L34-L38

Reply:
We know it throws if it doesn't wait because it's checked in the following test. It's the same case but every light is forced to green, and we expect a SimulationError when the train reaches the switch. 

An exception is maybe too much for getBranch, there could be legitimate use cases. I made it return null instead.